linera-service/src/benchmark.rs

```diff
@@ -72,46 +72,47 @@ async fn benchmark_with_fungible(
     uniform: bool,
 ) -> Result<()> {
     // Create the clients and initialize the wallets
-    let publisher = ClientWrapper::new(
-        Arc::new(tempdir().unwrap()),
-        Network::Grpc,
-        None,
-        num_wallets,
-    );
-    publisher.wallet_init(&[], Some(&faucet)).await.unwrap();
+    let dir = Arc::new(tempdir().context("cannot create temp dir")?);
+    let publisher = ClientWrapper::new(dir, Network::Grpc, None, num_wallets);
+    publisher.wallet_init(&[], Some(&faucet)).await?;
     let clients = (0..num_wallets)
-        .map(|n| ClientWrapper::new(Arc::new(tempdir().unwrap()), Network::Grpc, None, n))
-        .collect::<Vec<_>>();
-    join_all(clients.iter().map(|client| async {
-        client.wallet_init(&[], Some(&faucet)).await.unwrap();
-    }))
-    .await;
+        .map(|n| {
+            let dir = Arc::new(tempdir().context("cannot create temp dir")?);
+            Ok(ClientWrapper::new(dir, Network::Grpc, None, n))
+        })
+        .collect::<Result<Vec<_>, anyhow::Error>>()?;
+    try_join_all(
+        clients
+            .iter()
+            .map(|client| client.wallet_init(&[], Some(&faucet))),
+    )
+    .await?;
 
     // Sync their balances (sanity check)
-    join_all(clients.iter().map(|user| async move {
-        let chain = user.default_chain().unwrap();
-        let balance = user.synchronize_balance(chain).await.unwrap();
-        info!("User {:?} has {}", user.get_owner(), balance)
+    try_join_all(clients.iter().map(|user| async move {
+        let chain = user.default_chain().context("missing default chain")?;
+        let balance = user.synchronize_balance(chain).await?;
+        info!("User {:?} has {}", user.get_owner(), balance);
+        Ok::<_, anyhow::Error>(())
     }))
-    .await;
+    .await?;
 
     // Start the node services and subscribe to the publisher chain.
-    let publisher_chain_id = publisher.default_chain().unwrap();
-    let services = join_all(clients.iter().map(|client| async move {
-        let free_port = random_free_tcp_port().unwrap();
-        let chain_id = client.default_chain().unwrap();
-        let node_service = client.run_node_service(free_port).await.unwrap();
+    let publisher_chain_id = publisher.default_chain().context("missing default chain")?;
+    let services = try_join_all(clients.iter().map(|client| async move {
```

Comment:
I'm not sure what the performance repercussions would be here, but I suspect that you may get race conditions with the `random_free_port` once in a while if you do this in parallel. Sequential may be the way to go.

It's the setup phase anyway so it shouldn't effect the benchmark.